streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1014,6 +1015,23 @@ protected StreamsConfig(final Map<?, ?> props,
         if (props.containsKey(RETRIES_CONFIG)) {
             log.warn("Configuration parameter `{}` is deprecated and will be removed in 3.0.0 release.", RETRIES_CONFIG);
         }
+
+        if (eosEnabled) {
+            verifyEOSTransactionTimeoutCompatibility();
+        }
+    }
+
+    private void verifyEOSTransactionTimeoutCompatibility() {
+        final long commitInterval = (long) originals().getOrDefault(COMMIT_INTERVAL_MS_CONFIG, EOS_DEFAULT_COMMIT_INTERVAL_MS);
+        final long transactionTimeout = Long.valueOf((int) originals().getOrDefault(PRODUCER_PREFIX + ProducerConfig.TRANSACTION_TIMEOUT_CONFIG, DEFAULT_TRANSACTION_TIMEOUT));
```

Comment:
It might be simpler to just use `int transactionTimeout` -- Java will auto-cast to long in
```
if (transactionTimeout < commitInterval) {
```